docs/source/llms/prompt-engineering.rst

```diff
@@ -0,0 +1,341 @@
+Prompt Engineering UI (experimental)
+====================================
+
+Starting in MLflow 2.7, the MLflow Tracking UI provides a best-in-class experience for prompt
+engineering. With no code required, you can try out multiple LLMs from the
+:ref:`AI Gateway <gateway>`, parameter configurations, and prompts to build a variety of models for
+question answering, document summarization, and beyond. Using the embedded Evaluation UI, you can
+also evaluate multiple models on a set of inputs and compare the responses to select the best one.
+Every model created with the prompt engineering UI is stored in the :ref:`MLflow Model <models>`
+format and can be deployed for batch or real time inference. All configurations (prompt templates,
+choice of LLM, parameters, etc.) are tracked as :ref:`MLflow Runs <tracking>`.
+
+.. _prompt-engineering-quickstart:
+
+Quickstart
+==========
+
+The following guide will get you started with MLflow's UI for prompt engineering.
+
+Step 1: Create an AI Gateway Completions Route
+----------------------------------------------
+To use the prompt engineering UI, you need to create one or more :ref:`AI Gateway <gateway>`
+completions :ref:`Routes <routes>`. Follow the
+:ref:`AI Gateway Quickstart guide <gateway-quickstart>` to easily create a Route in less than five
+minutes. If you already have access to an AI Gateway Route of type ``llm/v1/completions``, you can
+skip this step.
+
+Step 2: Connect the AI Gateway to your MLflow Tracking Server
+-------------------------------------------------------------
+The prompt engineering UI also requires a connection between the AI Gateway and the MLflow
+Tracking Server. To connect the AI Gateway with the MLflow Tracking Server, simply set the
+``MLFLOW_AI_GATEWAY`` environment variable in the environment where the server is running and
+restart the server. For example, if the AI Gateway is running at ``http://localhost:7000``, you
+can start an MLflow Tracking Server in a shell on your local machine and connect it to the
+AI Gateway using the :ref:`mlflow server <cli>` command as follows:
+
+.. code-block:: bash
+
+   export MLFLOW_AI_GATEWAY="http://localhost:7000"
+   mlflow server --port 5000
+
+Step 3: Create or find an MLflow Experiment
+-------------------------------------------
+Next, open an existing MLflow Experiment in the MLflow UI, or create a new experiment.
+
+.. figure:: ../_static/images/experiment_page.png
+   :scale: 25%
+   :align: center
+
+Step 4: Create a run with prompt engineering
+--------------------------------------------
+Once you have opened the Experiment, click the **New Run** button and select
+*using Prompt Engineering*. This will open the prompt engineering playground where you can try
+out different LLMs, parameters, and prompts.
+
+|new_run| |prompt_modal_1|
+
+.. |new_run| image:: ../_static/images/new_run.png
+   :width: 25%
+
+.. |prompt_modal_1| image:: ../_static/images/prompt_modal_1.png
+   :width: 70%
```

Comment:
This part looks like this:

<img width="1640" alt="image" src="https://github.com/mlflow/mlflow/assets/17039389/20020dba-6246-421d-8273-66ab5894abd3">

Reply:
Yes, let me know if you have suggestions for improvements

Reply:
maybe GIF?

Reply:
or a screenshot like this:

<img width="1144" alt="image" src="https://github.com/mlflow/mlflow/assets/17039389/f6f3797b-8708-4da4-9c9a-807f62c3c691">